src/main.rs

```diff
@@ -69,21 +75,77 @@ impl Group {
 struct DWebBackend {
     path: String,
     port: u16,
+    store: Store,
+    veilid_api: Option<VeilidAPI>,
 }
 
 impl DWebBackend {
-    pub fn new(path: String, port: u16) -> Self {
-        DWebBackend { path, port }
-    }
+    pub fn new(base_path: &str, port: u16) -> Result<Self> {
+        let store_path = format!("{}/store.db", base_path);
+        let store = Store::persistent(&store_path).map_err(|e| anyhow!("Failed to create persistent store: {}", e))?;
+        Ok(DWebBackend {
+            path: base_path.to_string(),
+            port,
+            store,
+            veilid_api: None,
+        })
+    }
+
+    // Updated start method to initialize both Store and Veilid
     pub async fn start(&mut self) -> Result<()> {
-        // Init veilid
-        println!("Starting in {} with port {}", self.path, self.port);
+        println!("Starting on {} with port {}", self.path, self.port);
+
+        // Ensure base directory exists
+        let base_dir = &self.path;
+        fs::create_dir_all(base_dir).await.map_err(|e| {
+            anyhow!("Failed to create base directory {}: {}", base_dir, e)
+        })?;
+
+        // Initialize Veilid
+        let update_callback: Arc<dyn Fn(VeilidUpdate) + Send + Sync> = Arc::new(|update| {
+            info!("Received update: {:?}", update);
+        });
+
+        let mut rng = StdRng::from_entropy();
+        let random_suffix: u16 = rng.gen_range(10000..60000);
+
+        // Create a VeilidConfigInner instance
+        let config_inner = VeilidConfigInner {
+            program_name: format!("save-dweb-backend-{}", random_suffix),
```

Comment:
Is this suffix necessary? I feel like we should keep it stable between runs since it's the same program each time. Or maybe the suffix could be passed in the constructor? like `-cli` `-tests` `-android` `-ios`. Any clue where this string even gets exposed?